pkg/mirrorregistries/cluster_image_registry.go

```diff
@@ -0,0 +1,189 @@
+package mirrorregistries
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	RegistryConfKey                = "registries.conf"
+	RegistryCertKey                = "ca-bundle.crt"
+	imageRegistryName              = "additional-registry"
+	registryCertConfigMapName      = "additional-registry-certificate"
+	registryCertConfigMapNamespace = "openshift-config"
+	imageConfigMapName             = "additional-registry-config"
+	imageDigestMirrorSetKey        = "image-digest-mirror-set.json"
+	imageTagMirrorSetKey           = "image-tag-mirror-set.json"
+	registryCertConfigMapKey       = "additional-registry-certificate.json"
+	imageConfigKey                 = "image-config.json"
+)
+
+// GetImageRegistries reads a toml tree string with the structure:
+//
+// [[registry]]
+//
+//	 location = "source-registry"
+//
+//	   [[registry.mirror]]
+//		  location = "mirror-registry"
+//		  insecure = true # indicates to use an insecure connection to this mirror registry
+//		  pull-from-mirror = "tag-only" # indicates to add this to an ImageTagMirrorSet
+//
+// It will convert it to an ImageDigestMirrorSet CR and/or an ImageTagMirrorSet.
+// It will return these as marshalled JSON strings, and it will return a string list of insecure
+// mirror registries (if they exist)
+func GetImageRegistries(registryTOML string) ([]configv1.ImageDigestMirrors, []configv1.ImageTagMirrors, []string, error) {
```

Comment:
Can we deduplicate and of this with `ExtractLocationMirrorDataFromRegistriesFromToml`?